azkaban-common/src/main/java/azkaban/executor/ExecutorManager.java

```diff
@@ -98,12 +115,85 @@ public ExecutorManager(Props props, ExecutorLoader loader,
     executingManager = new ExecutingManagerUpdaterThread();
     executingManager.start();
 
+    queueProcessor = new QueueProcessorThread();
+    queueProcessor.start();
+
     long executionLogsRetentionMs =
-        props.getLong("execution.logs.retention.ms",
-            DEFAULT_EXECUTION_LOGS_RETENTION_MS);
+      props.getLong("execution.logs.retention.ms",
+        DEFAULT_EXECUTION_LOGS_RETENTION_MS);
     cleanerThread = new CleanerThread(executionLogsRetentionMs);
     cleanerThread.start();
+  }
 
+  /**
+   * <pre>
+   * Setup activeExecutors using azkaban.properties and database executors
+   * Note:
+   * 1. If a local executor is specified and it is missing from db,
+   *    this method add local executor as active in DB
+   * 2. If a local executor is specified and it is marked inactive in db,
+   *    this method will convert local executor as active in DB
+   * 3. If azkaban.use.multiple.executors is set true, this method will
+   *    load all active projects
+   * </pre>
+   *
+   * @throws ExecutorManagerException
+   */
+  public void setupExecutors() throws ExecutorManagerException {
```

Comment:
logic registering self as a new executor (in DB) I think is better  to be resided to the exeServer side  .  Here we should only load all the settings from the DB to keep the logic clean.